### DISCUSSION TITLE
Subscribe Topics For Non-HA Device

### DISCUSSION THREAD

**Discussion by Webbyz84** [briis/hass-weatherflow2mqtt#251] [Q&A]
Hello all, 

Firstly, pardon my lack of experience with MQTT and HA. I am learning. :) 

I was able to get a Mosquitto Container and this Container setup on my QNAP NAS and looking at the Terminal in the weatherflow container, it looks like it is publishing data to the MQTT Broker. 

My integration is a little different in that I am using a Tridium Niagara device to try to grab the MQTT data. I can tell that the MQTT Broker is working because I am able to use some various apps between my cell phone, the broker, and the Tridium Niagara software. I am using MyMQTT on my phone and publishing data to a topic on the broker, and the Tridium Niagara device is seeing that value and updating it. 

The issue that I am having is that I can't figure out the correct topic syntax to use to get the data that I need. Do I need to use the entire string like I see which is /homeassistant/sensor/weatherflow2mqtt_ST-XXXXXXXX/status/somethingelsehere to get the data? 

<img width="907" alt="SS2" src="https://github.com/user-attachments/assets/ac2f2f58-cb51-46c0-b1fc-8595d7224845" />
<img width="927" alt="SS1" src="https://github.com/user-attachments/assets/219d36f4-15f7-4dcc-b6ca-f0d4cf3b37c1" />

I am trying to various topics and can't seem to find the right one. 

Here is a screenshot of the software and some trial/errors. The first two points I am just using "air_temperature" and that is the point that I am able to update from my phone to the broker and the software gets the update. 
 
![Screenshot_3](https://github.com/user-attachments/assets/d49dbbe3-9138-48a2-9075-02fdaa9b8dbb)

Any help is appreciated!

**Comment by briis** [comment]
I have not been using this for a long time, so I can't really remember how the structure exactly is when publishing the data. I think to remember that there is a topic under each sensor type that contains a complete json list with all sensors that gives data. When I have time, I can try and set it up again, and see what I did.

Btw. I am also not maintaining this anymore, as this was done for Home Assistant, and there is now a core integration that does more or less what this did.

**Comment by Webbyz84** [comment]
Thank you for the reply. I know that what I am doing is outside of the
scope of your initial work. This just happens to be somewhat of a
side-project that I am trying to figure out and unfortunately I don't
really use Home Assistant. But the framework of what you have developed
would work for my application. I just can't seem to figure out the syntax
of the topics.

Any help is greatly appreciated and I again appreciate you taking the time
to reply in the first place.

**Comment by briis** [comment]
OK, so I did a bit of digging, and installed the Add-On on a Test Server.
I was right, all values are stored in 1 topic, in a JSON format. Let us say your Tempest Station is named ST-00123456. Then you will have a topic called:
`homeassistant/sensor/weatherflow2mqqt_ST-00123456/observation/state` and this topic will contain a value like the below:

```json
{"absolute_humidity": 5.98, "air_density": 1.28777, "air_temperature": 3.4, "battery": 2.71, "battery_level": 87, "battery_mode_description": "All sensors enabled and operating at full performance. Wind sampling interval every 3 seconds", "battery_mode": 0, "beaufort_description": "Light air", "beaufort": 1, "cloud_base": 89.0, "delta_t": 0.1, "dewpoint": 3.1, "dewpoint_description": "Somewhat Dry", "feelslike": 1.6, "freezing_level": 698.0, "illuminance": 227, "lightning_strike_count": 0, "lightning_strike_count_1hr": 0, "lightning_strike_count_3hr": 0, "lightning_strike_count_today": 0, "lightning_strike_distance": 0, "lightning_strike_energy": 0, "precipitation_type": "None", "rain_duration_today": 0, "rain_duration_yesterday": 0, "rain_intensity": "None", "rain_rate": 0.0, "rain_today": 0.0, "rain_yesterday": 0.0, "relative_humidity": 98.01, "sealevel_pressure": 1028.53, "solar_radiation": 2, "station_pressure": 1022.1, "temperature_description": "Cold", "uv": 0.0, "uv_description": "None", "visibility": 0.6, "wbgt": 1.3, "wetbulb": 3.2, "wind_bearing_avg": 273, "wind_direction_avg": "W", "wind_gust": 0.9, "wind_lull": 0.1, "wind_speed_avg": 0.4, "solar_elevation": -1, "solar_insolation": 0, "zambretti_number": 1, "zambretti_text": "Fine weather", "fog_probability": 87, "snow_probability": 0, "current_conditions": "fog", "pressure_trend": "Steady", "pressure_trend_value": 0, "last_reset_midnight": "2024-12-27T23:00:00+00:00"}
````


Beside from that, there is also a topic called `homeassistant/sensor/weatherflow2mqqt_ST-00123456/rapid_wind/state` which will have this data: 


```json
{"wind_speed": 0.5, "wind_bearing": 248, "wind_direction": "WSW"}
````

This is where you should be able to retrieve all the necessary data.
Did that help?
